### PR TITLE
Use the MATLAB Jet colormap for displaying depth data.

### PR DESCRIPTION
**Pull request by pchote**
Increases contrast by varying hue and saturation instead of just luminosity.

Uses the algorithm from https://github.com/kbinani/glsl-colormap.

### CODE REVIEW

File: glsl/color.frag [hunk 1]
```diff
@@ -1,6 +1,18 @@
 varying vec4 vColor;
 uniform bool EnableDepthPreview;
 
+float jet_r(float x) {
```
Comment (abcdefg30) [inline glsl/color.frag]: Should braces go on a new line here as well (we do it in the rest of this file, too)?

**Comment by Mailaender** [comment]
Doesn't work for me:

image

File: glsl/color.frag [hunk 2]
```diff
@@ -10,8 +22,11 @@ void main()
 
 	if (EnableDepthPreview)
 	{
-		// Front of the depth buffer is at 0, but we want to render it as bright
-		gl_FragColor = vec4(vec3(1.0 - gl_FragDepth), 1.0);
+		float x = 1.0 - gl_FragDepth;
+		float r = clamp(jet_r(x), 0.0, 1.0);
+		float g = clamp(jet_g(x), 0.0, 1.0);
+		float b = clamp(jet_b(x), 0.0, 1.0);
```
Comment (WolfGaming) [inline glsl/color.frag]: You sure you want to clamp this? And not do just a modulus?

Comment (pchote) [inline glsl/color.frag]: Quite sure.

**Comment by pchote** [comment]
Fixed braces.  This doesn't depend on #11904, but if you want the game to write non-zero depth data to the GPU then you do of course need to enable the depth buffer.

File: glsl/color.frag (hunk 2 above)
Comment (chrisforbes) [inline glsl/color.frag]: GL NDC convention has depth from -1 to 1, not 0 to 1. Are you sure you've got everything lined up how you want it, @pchote ?

Comment (pchote) [inline glsl/color.frag]: Pretty sure all the numbers are right.  I convert from NDC to window coords in the line above where the diff starts reporting.

**Comment by reaperrr** [comment]
Ingame:
colormap1

Makes it a bit less hard to tell things apart (though I still find myself using pixel-doubling when checking depth data).
:+1:

**Comment by reaperrr** [comment]
@pchote I guess now that #11904 is merged, you could rebase this so the next reviewer doesn't have to enable depth manually, but that's up to you.

**Comment by obrakmann** [comment]
fine with me, :+1:

**Pull request merged**